examples/controller.js

```diff
@@ -1,6 +1,41 @@
-module.exports = function($scope, GroupList) {
-	$scope.insightOptions = {
-		data: GroupList.groups
-	};
-	$scope.selectedGroups = [GroupList.groups[1], GroupList.groups[2]];
+var _ = require('underscore');
+
+module.exports = function($scope, GroupList, MovieService) {
+	_.extend($scope, {
+		basicOptions: {
+			data: GroupList.groups
+		},
+		assignedGroups: [GroupList.groups[1], GroupList.groups[2]],
+
+		movieOptions: {
+			data: [],
+			loadPage: function (query, deferred) {
+				if (!query.length) {
+					return deferred.resolve();
+				}
+
+				MovieService
+					.queryMovies(query)
+					.then(function (movies) {
+						//rename title property to name until display properties are supported
+						//Call deferred.resolve with result array. Replaces results already stored by insight.
+						deferred.resolve(movies
+							.map(function (movie) {
+								return {
+									id: movie.id,
+									name: movie.title
+								};
+							})
+							.map(function (movie) {
+								var existingMovie = _.findWhere($scope.movieOptions.data, movie);
+								return existingMovie || movie;
```

Comment:
This second map function replaces any items stored locally by the directive with the local object. This operation is required to display search results as assigned in the overlay.

We should think more about whether or not this should be included within the directive. For now, it's probably best to keep this as is and limit the assumptions made by the directive about the data.